src/environment/ale_state.cpp

```diff
@@ -259,6 +259,19 @@ void ALEState::applyActionPaddles(Event* event, int player_a_action, int player_
   }
 }
 
+void ALEState::pressSelect(Event* event) {
+  resetKeys(event);
+  event->set(Event::ConsoleSelect, 1);
+}
+
+void ALEState::setDifficulty(Event* event, unsigned mask) {
+  resetKeys(event);
+  event->set(Event::ConsoleLeftDifficultyA, mask & 1);
```

Comment:
If I understand correctly, this scheme assumes that 'event' remains the same throughout -- i.e. the difficulty bits never get cleared except through this method. Is this going to be an issue for serialization?

Reply:
To be honest, I can't say for sure.
I think it would be better if we went through with this now and then we check serialization, for example. Since more has to be changed in order to get this feature, I'd rather do this and then change the code accordingly if serialization becomes a problem.